packages/xdl/src/Api.ts

```diff
@@ -87,19 +88,22 @@ async function _callMethodAsync(
     };
   }
 
-  if (requestOptions) {
-    if (requestOptions.formData) {
-      let convertedFormData = await _convertFormDataToBuffer(requestOptions.formData);
-      let { data } = convertedFormData;
-      options.headers = {
-        ...options.headers,
-        ...requestOptions.formData.getHeaders(),
-      };
-      options = { ...options, data };
-    } else {
-      options = { ...options, ...requestOptions };
-    }
+  if (!requestOptions.timeout && ConnectionStatus.isOffline()) {
```

Comment:
```suggestion
  if (!requestOptions.hasOwnProperty('timeout') && ConnectionStatus.isOffline()) {
```

this is important because if `timeout` is `0` then `!0` also returns `true`, so people wouldn't be able to set `timeout` to 0.

Reply:
to be clear on the semantics - if the connection is offline then we are saying we will still respect the timeout property if it is provided

Reply:
0 means no timeout, though. Should we allow that if they're offline? That would result in an endless hang

Reply:
i think if someone passes in a timeout we should respect the option, and it is up to the caller to decide if they want to change the timeout depending on the connection status